### DISCUSSION TITLE
Installing GNU compilers on Windows

### DISCUSSION THREAD

**Discussion by AliSSDL** [nrc-cnrc/EGSnrc#763] [Questions]
How can I install mingw32 on Windows?
```
C:\Users\ACC>mingw32-make --version 
mingw32-make  is not recognized as an internal or external command.
```

**Comment by ftessier** [comment]
@AliSSDL Please consider and follow the instructions to install `mingw-w64`, which is distributed separately: http://mingw-w64.org/

**Reply by ftessier** [reply]
@AliSSDL See also the [notes regarding mingw](https://github.com/nrc-cnrc/EGSnrc/wiki/Install-EGSnrc-on-Windows#obtaining-and-setting-up-the-gnu-compilers) in the wiki.

**Reply by AliSSDL** [reply]
I have some problems in icon view (egs-inprz), I can't restore to a 
![SharedScreenshot (2)](https://user-images.githubusercontent.com/89786824/149614784-874af096-43fb-4595-b4b9-e6c5d11432ac.jpg)
small (minimize ) view?

**Comment by Jaap1957** [comment]
Hi, did you manage to get EGSnrc working?

The above problem might be because you did not set the PATH  to include mingw.

**Reply by AliSSDL** [reply]
Hi jaap , many thanks for your reply . Didn't get EGSnrc yet but I
Still Have same problem .

**Reply by rtownson** [reply]
Did you install mingw and update your environment variables?

**Reply by AliSSDL** [reply]
yes, I follow all the steps.

**Reply by AliSSDL** [reply]
did someone have a youtube link showing the processing of installation?

**Reply by rtownson** [reply]
> did someone have a youtube link showing the processing of installation?

There is a windows installation video on this channel: 
https://www.youtube.com/channel/UC_GN5tmb7Kah93_lrJ62n9w/videos

**Reply by AliSSDL** [reply]
many thanks

**Comment by Jaap1957** [comment]
Is this still the situation?

`C:\Users\ACC>mingw32-make --version`
`mingw32-make  is not recognized as an internal or external command.`

Find the path to mingw32-make.

`c:\users\ACC>where mingw32-make`

this should produce the path to the executable, for example:

`c:\Program Files\mingw-w64\x86_64-7.3.0-posix-seh-rt_v5-rev0\mingw64\bin\mingw32-make.exe`

What is the result at your pc?

**Reply by AliSSDL** [reply]
the path is C:\users\Ali\ !!!

**Reply by rtownson** [reply]
> the path is C:\users\Ali\ !!!

Can you include screenshots of the file browser showing mingw32-make.exe, and showing your PATH environment variable, which should contain the path to mingw? Maybe on seeing these screenshots we will spot something.

**Reply by AliSSDL** [reply]
![DSC_0251_3](https://user-images.githubusercontent.com/89786824/149378442-869f67fe-7c07-40eb-b1ca-4228b3148e71.JPG)

**Reply by Jaap1957** [reply]
There can be more than 1 mingw32-make.exe on your PC.

with the command
`dir /s /b c:\mingw32-make.exe`
you can find how many.

with 
`where mingw32-make.exe`
you can find out which one is in your PATH.

**Reply by rtownson** [reply]
This error might mean you have installed 32-bit mingw on a 64-bit machine, or visa versa.

**Reply by AliSSDL** [reply]
many thanks, I think passing these steps is right?  I follow prerequisite and youtube 
![DSC_0023](https://user-images.githubusercontent.com/89786824/149385032-0db34896-6c39-4a33-9c01-547fda322061.JPG)

**Reply by Jaap1957** [reply]
looks good !

**Reply by AliSSDL** [reply]
many thanks

**Reply by AliSSDL** [reply]
many many thanks all
![DSC_0027](https://user-images.githubusercontent.com/89786824/149413281-4be1b04b-6245-46a5-b5ac-4636cfacb189.JPG)
.

**Comment by AliSSDL** [comment]
can one help me to compile my app for tutorial 7? failed in first

![SharedScreenshot (3)](https://user-images.githubusercontent.com/89786824/150568602-9a534b06-eb09-43b5-ad91-0827a11f676f.jpg)
 step?

**Comment by Jaap1957** [comment]
cd. with the dot right after cd is not a command
cd ..\  with a space behind cd is a command

________________________________

**Reply by AliSSDL** [reply]
PS C:\Users\Ali> cd.. \egs_home\myapp
PS C:\Users> \egs_home\mayapp>mingw32.make.exe
\egs_home\mayapp>mingw32.make.exe : The term '\egs_home\mayapp>mingw32.make.exe' is not recognized as the name of a
cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify
that the path is correct and try again.
At line:1 char:1
+ \egs_home\mayapp>mingw32.make.exe
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (\egs_home\mayapp>mingw32.make.exe:String) [], CommandNotFoundException
    + FullyQualifiedErrorId : CommandNotFoundException

**Reply by rtownson** [reply]
Please complete some beginner tutorials for the command prompt, so that you can understand the errors for yourself.

**Reply by AliSSDL** [reply]
many thanks

**Comment by Jaap1957** [comment]
mingw32-make
use a minus - between mingw32 and make, not a dot .

________________________________

**Reply by AliSSDL** [reply]
![SharedScreenshot (4)](https://user-images.githubusercontent.com/89786824/150595999-e1585878-0a9b-4645-b41e-f19c69a7fe74.jpg)

**Reply by Jaap1957** [reply]
Hi, it should look like this 
![afbeelding](https://user-images.githubusercontent.com/96918927/150643095-87ef9467-6fa2-4564-a487-92781a7a0410.png)

So the program is not found.
Then, it is not in the PATH.

![afbeelding](https://user-images.githubusercontent.com/96918927/150643161-c6152aed-0c96-4b8f-bc43-a99f1262bc96.png)

If this produces a large blurb of test. mingw will be in it somewhere.
If this returns nothing: you should add mingw to your path environment variable.

Here how to add some folder to the path.

https://helpdeskgeek.com/windows-10/add-windows-path-environment-variable/

ming232-make.exe --version worked a few days ago . Try that again. What was changed?

**Reply by AliSSDL** [reply]
many thanks, I will try

**Reply by rtownson** [reply]
One of the problems was that the 'cd' command was not used correctly, there was an extra '..'.

**Reply by AliSSDL** [reply]
still the same problem?

**Reply by rtownson** [reply]
Did you work through command line tutorials? Example:
```
cd EGSnrc\egs_home\myapp
mingw32-make.exe
```

**Reply by AliSSDL** [reply]
no,
